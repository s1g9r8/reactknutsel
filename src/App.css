
:root {
  --main-color: #768d7d;
}

h1 {
  width: 100%;
  color: var(--main-color);
}

.header {
  background-color: white;
  box-shadow: 0 0 15px 1px rgba(0,0,0,.1);
  padding-top: 25px;
  padding-bottom: 25px;
}

.navigation a {
  color: var(--main-color);
  text-decoration: none;
  margin-right: 30px;
  padding-bottom: 5px;

}

.navigation a.active {
  border-bottom: 2px solid var(--main-color);
}

.navigation a:hover {
  border-bottom: 2px solid var(--main-color);
  transition: ease-in-out .2s;
}

.content {
  margin-top: 30px;
}

.project-page {
  display: flex;
  flex-wrap: wrap;
}

.project-list {
  width: 70%;
}

.project-form {
  width: 30%;
}

.project-form-item {
  display: flex;
  margin-bottom: 10px;
  align-items: flex-start;
}

.project-form-item label {
  width: 35%;
}

.project-form-item > span {
  width: 65%;
}

.project-form-item span input {
  width: 100%
}
/*.project-form-item label {
  width: 50%;
}

.project-form-item input {
  width: 50%;
}*/
