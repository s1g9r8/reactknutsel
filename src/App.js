
import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import bootstrap from 'bootstrap/dist/css/bootstrap.css';

import Nav from './components/common/Nav.js';
import PlanningPage from './components/planning/PlanningPage.js';
import EmployeesPage from './components/employees/EmployeesPage.js';
import ProjectPage from './components/projects/ProjectPage.js';
//import './App.css';
import './stylesheets/styles.css';

const App = () => {
  return (
    <div>
      <Router>
        <div className="app">
          <Nav />
            <div className="content container">
              <Route exact path="/" component={PlanningPage}/>
              <Route path="/employees" component={EmployeesPage}/>
              <Route path="/projects" component={ProjectPage}/>
            </div>
        </div>
      </Router>
    </div>
  );
};

export default App;
