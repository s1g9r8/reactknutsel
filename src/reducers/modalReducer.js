
var modalReducer = (state = [], action) => {
  switch (action.type) {
    case 'SET_MODAL':
      return action.modal;
    default:
      return state;
  }
};

export default modalReducer;
