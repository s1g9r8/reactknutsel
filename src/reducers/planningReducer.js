import uuid from 'uuid';
import moment from 'moment';

var planningReducer = (state = [], action) => {

  var weekNumber = (state.length ? (state[state.length - 1].weekNumber + 1) : moment().week())

  //TODO: make year as a variable and check if the weeknumber is not larger than largest number in year

  switch (action.type) {
    // case 'ADD_WEEK':
    //   return [
    //     ...state,
    //     {
    //       id: uuid(),
    //       weekNumber: weekNumber,
    //       year: 2017,
    //       projects: 'projects'
    //     }
    //   ];
      case 'GET_WEEKS':
        return {
          'i1' : {weekNumber: 36,
                  year: 2017,
                  projects: { 'p1': { 'e1': {name: 'suzanne', hours: 36}, 'e2': {name: 'olger', hours: 40} },
                              'p2': { 'e3': {name: 'her', hours: 40}, 'e4': {name: 'danny', hours: 36} }
                            }
          },
          'i2' : {weekNumber: 37,
                  year: 2017,
                  projects: { 'p1': { 'e1': {name: 'suzanne', hours: 36}, 'e2': {name: 'olger', hours: 40} },
                              'p2': { 'e3': {name: 'her', hours: 40}, 'e4': {name: 'danny', hours: 36} }
                            }
          }
      }
      case 'ADD_PROJECT':
          return {...state, concept: true};
    default:
      return state;
  }
};

export default planningReducer;
