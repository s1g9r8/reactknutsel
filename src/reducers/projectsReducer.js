
var projectsReducer = (state = {}, action) => {
  switch (action.type) {
    case 'ADD_PROJECTS':
      return action.projects
    case 'ADD_PROJECT':
      return {...state, concept: true};
    case 'UPDATE_PROJECT_CONCEPT_STATUS': {
      var p = state[action.id];
      p.concept = action.concept;
      return {...state, [action.id]: p};
    }
    case 'UPDATE_PROJECT':
      return {...state, [action.id]: action.project}
    case 'VALIDATE_PROJECT':
      return state.map((project) => {
        if (project.id === action.id) {
          return {
            ...project,
            nameErrorText: action.nameErrorText,
            customerErrorText: action.customerErrorText,
            startDateErrorText: action.startDateErrorText
          };
        } else {
          return project;
        }
      });
    default:
      return state;
  }
};

export default projectsReducer;
