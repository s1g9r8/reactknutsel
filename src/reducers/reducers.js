import uuid from 'uuid';
import moment from 'moment';

export var weekListReducer = (state = [], action) => {

  var weekNumber = (state.length ? (state[state.length - 1].weekNumber + 1) : moment().week())

  //TODO: make year as a variable and check if the weeknumber is not larger than largest number in year

  switch (action.type) {
    case 'ADD_WEEK':
      return [
        ...state,
        {
          id: uuid(),
          weekNumber: weekNumber,
          year: 2017
        }
      ];
    case 'UPDATE_WEEK':
      return state.map((week) => {
        if (week.id === action.id) {
          return {
            ...week,
            monday: action.week.monday,
            tuesday: action.week.tuesday,
            wednesday: action.week.wednesday,
            thursday: action.week.thursday,
            friday: action.week.friday
          };
        } else {
          return week;
        }
      });
    default:
      return state;
  }
};
