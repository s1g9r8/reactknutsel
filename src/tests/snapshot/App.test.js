import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore, compose } from 'redux';

import App from './../../App.js';
import { weekListReducer } from './../../reducers/reducers.js';


var store = createStore(weekListReducer, {});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><App /></Provider>, div);
});


it('renders a snapshot', () => {
  const tree = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
