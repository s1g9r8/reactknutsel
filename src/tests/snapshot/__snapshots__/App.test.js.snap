// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <div>
    <a
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Planning
    </a>
    <a
      className={undefined}
      href="/employees"
      onClick={[Function]}
      style={undefined}
    >
      Employees
    </a>
    <a
      className={undefined}
      href="/projects"
      onClick={[Function]}
      style={undefined}
    >
      Projects
    </a>
  </div>
  <div>
    <div />
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        add week
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        calculate
      </button>
    </div>
  </div>
</div>
`;
